pkg/aggregator/check_sampler.go

```diff
@@ -32,11 +34,18 @@ func newCheckSampler(hostname string) *CheckSampler {
 func (cs *CheckSampler) addSample(metricSample *metrics.MetricSample) {
 	contextKey := cs.contextResolver.trackContext(metricSample, metricSample.Timestamp)
 
-	cs.metrics.AddSample(contextKey, metricSample, metricSample.Timestamp, 1)
+	if err := cs.metrics.AddSample(contextKey, metricSample, metricSample.Timestamp, 1); err != nil {
+		log.Debug("Ignoring sample '%s' on host '%s' and tags '%s': %s", metricSample.Name, metricSample.Host, metricSample.Tags, err)
+	}
 }
 
 func (cs *CheckSampler) commit(timestamp float64) {
-	for _, serie := range cs.metrics.Flush(timestamp) {
+	series, errors := cs.metrics.Flush(timestamp)
+	for ckey, err := range errors {
+		context := cs.contextResolver.contextsByKey[ckey]
```

Comment:
check return value?  Otherwise log message below might crash.

Reply:
good point indeed. In no circumstances should the context key not be tracked by the `contextResolver` (unless something's horribly wrong in the aggregator's logic), that said we should still handle this case. Applies to the existing logic below too. I'll change this.